packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -802,4 +807,48 @@ export const logExperimentDeleted = async (
   return emittedEvent.id;
 };
 
-// endregion Events
+export const getAllVisualExperiments = async (
+  organization: string
+): Promise<
+  Array<{
+    experiment: ExperimentInterface;
+    visualChangeset: VisualChangesetInterface;
+  }>
+> => {
+  const visualChangesets = await findVisualChangesets(organization);
+  const visualChangesByExperimentId = visualChangesets.reduce(
+    (acc: Record<string, Array<VisualChange>>, c) => {
+      if (!acc[c.experiment]) acc[c.experiment] = [];
+      acc[c.experiment] = acc[c.experiment].concat(c.visualChanges);
+      return acc;
+    },
+    {}
+  );
```

Comment:
We should have an early return if there aren't any visual change sets. No reason to run the extra experiments Mongo query, which can be expensive

Reply:
Added